test/src/String/test_comparisonFunc.cpp

```diff
@@ -19,25 +19,33 @@
 TEST_CASE ("Testing String::equals(const String &) with exit status PASS", "[String-equals-01]")
 {
   arduino::String str1("Hello"), str2("Hello");
-  REQUIRE(str1.equals(str2) == 1);
+  CHECK(str1.equals(str2) == 1);
```

Comment:
According to the Catch2 documentation ...
> The REQUIRE family of macros tests an expression and aborts the test case if it fails. The CHECK family are equivalent but execution continues in the same test case even if the assertion fails.

I can see what you are doing it but its not optimal test code design. Ideally there should be only one assertion per test case, if there is more than one assertion it is often a sign that you are testing different things at the same time at should split the "concept under test" into multiple test functions. Right here you are testing `String::equals`, `String::operator ==` and `String::operator !=`. Please consider refactoring the test code accordingly.

Reply:
Note: Consequently it's preferable to use `REQUIRE` over `CHECK` macros.

Reply:
This commit is really the subject of #137, where this issue was discussed and solved, so I'm resolving this comment.